float64[] q
float64[] qdot
float64[] f
